port/action/schema.go

```diff
@@ -111,64 +127,193 @@ func ActionSchema() map[string]schema.Attribute {
 			MarkdownDescription: "Description",
 			Optional:            true,
 		},
-		"required_approval": schema.BoolAttribute{
-			MarkdownDescription: "Require approval before invoking the action",
-			Optional:            true,
-		},
-		"approval_webhook_notification": schema.SingleNestedAttribute{
-			MarkdownDescription: "The webhook notification of the approval",
+		"self_service_trigger": schema.SingleNestedAttribute{
+			MarkdownDescription: "Self service trigger for the action",
 			Optional:            true,
 			Attributes: map[string]schema.Attribute{
-				"url": schema.StringAttribute{
-					MarkdownDescription: "The URL to invoke the webhook",
+				"blueprint_identifier": schema.StringAttribute{
+					Description: "The ID of the blueprint",
+					Optional:    true,
+				},
+				"operation": schema.StringAttribute{
+					MarkdownDescription: "The operation type of the action",
 					Required:            true,
+					Validators: []validator.String{
+						stringvalidator.OneOf("CREATE", "DAY-2", "DELETE"),
+					},
+				},
+				"user_properties": schema.SingleNestedAttribute{
+					MarkdownDescription: "User properties",
+					Optional:            true,
+					Attributes: map[string]schema.Attribute{
+						"string_props":  StringPropertySchema(),
+						"number_props":  NumberPropertySchema(),
+						"boolean_props": BooleanPropertySchema(),
+						"object_props":  ObjectPropertySchema(),
+						"array_props":   ArrayPropertySchema(),
+					},
+				},
+				"required_jq_query": schema.StringAttribute{
+					MarkdownDescription: "The required jq query of the property",
+					Optional:            true,
+				},
+				"order_properties": schema.ListAttribute{
+					MarkdownDescription: "Order properties",
+					Optional:            true,
+					ElementType:         types.StringType,
 				},
 			},
+			Validators: []validator.Object{
+				objectvalidator.ExactlyOneOf(
+					path.MatchRoot("self_service_trigger"),
+					path.MatchRoot("automation_trigger"),
+				),
+			},
 		},
-		"approval_email_notification": schema.ObjectAttribute{
-			MarkdownDescription: "The email notification of the approval",
+		"automation_trigger": schema.SingleNestedAttribute{
+			MarkdownDescription: "Automation trigger for the action",
```

Comment:
let's add a note regarding that this schema expects only one of the attributes to be defined